### ISSUE TITLE
Migrate to "MDC" Material Components

### ISSUE THREAD

**Issue by danoswaltCL** [CarnegieLearningWeb/UpGrade#1196]
This is a requirement before we can move to Angular 17:

https://rc.material.angular.io/guide/mdc-migration

This will require running a script:
`ng generate @angular/material:mdc-migration`

And then taking care of the "TODOs" that it creates.
This will likely involve some re-styling and potentially some code changes to accommodate stuff that script can't fix for us (hint: there's quite a bit).

Dependencies:
This should wait until Angular 16 is merged into dev ##1189 is completed

AC:
The application will look and feel as close to unchanged as we can.
